### ISSUE TITLE
Include content length in header when downloading file so that users can display progress

### ISSUE THREAD

**Issue by ameer-clara** [ethersphere/bee-js#525]
bee-js version: 3.1.0

When downloading files using the bee client the progress event `total` is always 0.

Example code 

```

async download(reference: any) {
    let result = undefined;

    const logProgress = (input: any) => {
      console.log("progress: ", input);
      };
    const axiosInstance = axios.create({
      onDownloadProgress: logProgress,
    });
    const fetch = buildAxiosFetch(axiosInstance);
    try {
      let bundle = await this.bee.downloadFile(reference, undefined, { fetch });
      result = bundle.data.json();
    } catch (e) {
      console.log("error downloading", e);
      result = e;
    }
    return result;
  }
```

Example callback object logs:

Download progress event 1

{
	isTrusted: true
	bubbles: false
	cancelBubble: false
	cancelable: false
	composed: false
	defaultPrevented: false
	eventPhase: 0
	lengthComputable: false
	loaded: 39952
	path: []
	returnValue: true
	timeStamp: 2811.600000023842
	**total: 0**
	type: "progress"
}

event 5

{
	isTrusted: true
	bubbles: false
	cancelBubble: false
	cancelable: false
	composed: false
	defaultPrevented: false
	eventPhase: 0
	lengthComputable: false
	loaded: 601114
	path: []
	returnValue: true
	timeStamp: 3482.600000023842
	**total: 0**
	type: "progress"
}
Is it possible to add a header, X-Actual-Content-Length, with the content size? So that the users can be shown a progress bar and know how long it will take for the content to load.

* note when uploading the total property gets updated

**Comment by agazso** [comment]
Thanks for reporting this issue! We will investigate it.

**Comment by AuHau** [comment]
Thanks for reporting!

I have investigated it and there is a combination of factors happening here. Even though Bee properly sets `Content-Length` header, since browsers are using `gzip` compression to fetch the data, they are not exposing the `Content-Length` in the response and hence the `total` is zero.

I have also noticed that Bee actually sets a custom header called `Decompressed-Content-Length`, but this is not exposed to the Response object because of CORS settings. I have created fix for this in above linked PR and once merged and released (maybe for the upcoming 1.4.2 version? not sure) this should be possible with getting the header as `e.target.getResponseHeader("Decompressed-Content-Length")`.

**Comment by ameer-clara** [comment]
Wow, that's fast! Thanks for the fix and providing the details. Can't wait to try the new version.

**Comment by tfius** [comment]
encountered this problem with binary files where stream readers expect proper content-lenght, is this merged in 1.4.2 ?

**Comment by AuHau** [comment]
It was merged to master, but I am not sure if it makes it to 1.4.2. Btw. this won't fix `content-length` but it will expose the info in different header.

**Comment by tfius** [comment]
Hmm, could `content-length` be injected at least in gateways

**Comment by AuHau** [comment]
This is limitation of browsers (or at least Chrome's) not ours. 

> I have investigated it and there is a combination of factors happening here. Even though Bee properly sets Content-Length header, since browsers are using gzip compression to fetch the data, they are not exposing the Content-Length in the response and hence the total is zero.

See also this issue: https://github.com/axios/axios/issues/1591#issuecomment-431400903

**Comment by AuHau** [comment]
So the fix in Bee should land in 1.4.2 in matter of days (or maybe even today).

**Comment by agazso** [comment]
The fix is indeed released in 1.4.2:
https://github.com/ethersphere/bee/releases/tag/v1.4.2

**Comment by AuHau** [comment]
I am closing this as this should be resolved. Please let me know if it is not.
